tsfc/driver.py

```diff
@@ -329,22 +370,68 @@ def compile_expression_at_points(expression, points, coordinates, interface=None
     expression = ufl_utils.split_coefficients(expression, builder.coefficient_split)
 
     # Translate to GEM
-    point_set = PointSet(points)
-    config = dict(interface=builder,
-                  ufl_cell=coordinates.ufl_domain().ufl_cell(),
-                  precision=parameters["precision"],
-                  point_set=point_set,
-                  argument_multiindices=argument_multiindices)
-    ir, = fem.compile_ufl(expression, point_sum=False, **config)
-
-    # Deal with non-scalar expressions
-    value_shape = ir.shape
-    tensor_indices = tuple(gem.Index() for s in value_shape)
-    if value_shape:
-        ir = gem.Indexed(ir, tensor_indices)
+    kernel_cfg = dict(interface=builder,
+                      ufl_cell=coordinates.ufl_domain().ufl_cell(),
+                      precision=parameters["precision"],
+                      argument_multiindices=argument_multiindices,
+                      index_cache={})
+
+    if len(to_element.value_shape()) == 0:
```

Comment:
`len(to_element.value_shape()) == 0` refers to the `to_element` being scalar valued, which is different than all the nodes of the element being point-evaluation.  (Example, if someone wrote a Lagrange-like element but with integral moments over edges.

Reply:
Would something like this be safer?
```suggestion
    only_pointeval = all(isinstance(dual, FIAT.functional.PointEvaluation) for dual in to_element.dual_basis())
    if only_pointeval:
```

Reply:
Should we commit this change?

Reply:
Yes.